addons/html_editor/static/src/utils/sanitize.js

```diff
@@ -16,7 +17,22 @@ export function initElementForEdition(element, options = {}) {
     }
 }
 
+/**
+ * Properly close common XML-like self-closing elements to avoid HTML parsing
+ * issues.
+ *
+ * @param {string | Markup} content
+ * @returns {string | Markup}
+ */
 export function fixInvalidHTML(content) {
+    if (!content) {
+        return content;
+    }
+    // TODO: improve the regex to support nodes with data-attributes containing
+    // `/` and `>` characters.
     const regex = /<\s*(a|strong|t)[^<]*?\/\s*>/g;
-    return content.replace(regex, (match, g0) => match.replace(/\/\s*>/, `></${g0}>`));
+    const fixedContent = content.replace(regex, (match, g0) =>
+        match.replace(/\/\s*>/, `></${g0}>`)
+    );
+    return content.constructor?.name === "Markup" ? markup(fixedContent) : fixedContent;
```

Comment:
Can't we compare the actual constructor function rather than the name?

Reply:
@dmo-odoo No, because unfortunately, owl does not export `Markup` :/